Add RabbitMQ prefetch support

mihaitodor opened the pull request:
Add support for specifying the RabbitMQ prefetch value, which I left as a TODO [here](https://github.com/upfluence/sensu-go/blob/389afcd82a33b6dfd753bcea790c6c584fe07060/sensu/transport/rabbitmq/rabbitmq.go#L54).

AlexisMontagne commented on sensu/transport/rabbitmq/rabbitmq.go:
```diff
@@ -4,19 +4,60 @@ import (
 )
 
+// AmqpChannel is an interface over amqp.Channel
+type AmqpChannel interface {
```
because amqp is an acronym, the Go styleguide recommend to name it all in capital, `AMQPChannel`

AlexisMontagne commented on sensu/transport/rabbitmq/rabbitmq.go:
```diff
@@ -95,6 +157,25 @@ func (t *RabbitMQTransport) Connect() error {
 		return err
 	}
 
+	if prefetchString := config.Prefetch.String(); prefetchString != "" {
```
Instead of the `.String()` method, can we use the `.Int64()` to avoid the explicit string conversion?

mihaitodor replied:
Actually, that won't work properly, because `.Int64()` returns an `invalid syntax` error when `config.Prefetch == ""` and I wanted to avoid logging a warning in this case since that's a legit value if we don't specify the prefetch value in the config.

AlexisMontagne replied:
You're right!

AlexisMontagne commented on sensu/transport/rabbitmq/rabbitmq.go:
```diff
@@ -95,6 +157,25 @@ func (t *RabbitMQTransport) Connect() error {
+		// https://github.com/ruby-amqp/amqp/blob/9880a2b5dcfe4b27cefbdb3b3e2ea3ec58ea348a/lib/amqp/channel.rb#L1214
+		err = t.Channel.Qos(prefetch, 0, false)
+
+		if err != nil {
```
i think you can squash the if we the previous  line, ` if err := t.Channel.Qos(prefetch, 0, false); err != nil {`

AlexisMontagne commented on sensu/transport/rabbitmq/rabbitmq.go:
```diff
@@ -95,6 +157,25 @@ func (t *RabbitMQTransport) Connect() error {
+		prefetch, err = strconv.Atoi(prefetchString)
+
+		if err != nil {
+			log.Warningf("Failed to parse the prefetch value \"%s\": %s", err.Error())
```
maybe we should just continue, instead of doing a (useless?) QoS call

mihaitodor replied:
Nice catch!

AlexisMontagne reviewed (commented):
Really good contribution overall! 👍 

I just gave you some feedbacks about syntax

mihaitodor committed:
Add RabbitMQ prefetch support

mihaitodor committed:
Fix linter issue: Connection.Close() can return an error

mihaitodor committed:
Add prefetch tests

mihaitodor committed:
Fix review comment: avoid doing a useless Qos() call

mihaitodor committed:
Fix linter warning: omit comparison to bool constant

mihaitodor commented:
@AlexisMontagne I have rebased the change set and addressed your comments, except [one](https://github.com/upfluence/sensu-go/pull/5#discussion_r96641219). Please let me know if you spot anything else.

AlexisMontagne commented:
Perfect, the code looks great now, i'll merge it
